### DISCUSSION TITLE
PMGH is shutting down

### DISCUSSION THREAD

**Discussion by afkvido** [Prodigy-Hacking/ProdigyMathGameHacking#1341] [(A) Anything]
# PMGH is shutting down

## Chapter 1: PMGH is stopping it's service.

On March 26, 2022, at 9:17 AM PST, @BoredFishRE (`BoredFish#4269`) said this:

```markdown
Important message for @everyone:

First we'd like to thank everyone for being with us. For the past 2.5 years, we've maintained hacks for the math game Prodigy.

Our original goal, which was raising awareness about the security of Prodigy has (mostly) been accomplished!
While we would love to continue maintaining hacks, we collectively have had less and less time to focus on development.
Server costs are also surprisingly expensive..! Who knew?

For these reasons, As of **April 10th, 2022**, we will shut down our instance of P-NP.
**Hacks for the most part will no longer work.**

We will keep our source code on GitHub, where anyone can continue development by forking our repositories.
We'll continue to review pull requests, but patched hacks most likely will remain patched.

If you would like to run your own instance of the hacks, we'll soon release instructions on how to run P-NP.
This discord server will move from a Prodigy Hacking server to a Coding focused server over the next month.

Thank you all for understanding, and being with us for the past few years.
So long, and thanks for all the fish.
```

## Chapter 2: Maybe Not?

Speaking of **time**, while the official PMGH team doesn't have the time to maintain hacks... new contributors could be accepted! What about Pull Requests?

Speaking of **money** (server costs), PMGH can start a Gofundme, Patreon, PayPal, or literally ANY donation system. Why does PMGH not accept donations in the first place? PMGH is literally a bunch of young guys who go to college, there's no shame in taking donations.

And if my two ideas aren't possible and realistic, then let's see Chapter 3.

## Chapter 3: A last stand.

First of all, (I know that this would make the hacks an actual pain in the ass to update), but **how about we put P-NP into the extension itself?**

An even better idea though is to find someone who can host P-NP for PMGH though. (or just start a goddamn gofundme).

### Chapter 3a: Raid Prodigy?

...Also something I'd like to do before P-NP goes offline is try to raid the Prodigy servers by blocking all NPCs with the Morph Hitbox trick.

1. Click the hack: `Morph Player`
2. Select the Morph Type: `Follow`
3. Select the Morph: `Horned Owl`
4. You'll now appear bugged, as an image file that stores sprites.
5. Anyways, that image has a massive hitbox, and about 5 of those can cover an entire zone, making it unclickable, they'll only be able to open your profile.
6. Keep in mind that this doesn't work on Harmony Island.

## Chapter 4: The Discord Server

BoredFish mentioned turning the Discord Server into a coding server.
I'm asking: "Wait, what the hell will happen to the legacy of PMGH?"

The `Prodigy Hacking` Discord Server has so much history to it, I know that the people in the server will be the same, but the discussion in the server might rapidly change.

I know that this change will happen and there's nothing I can do about it. I don't **hate** this change, I'm just a bit worried about the culture of the server changing.

## Chapter 5: This Repository

While the hacks might stop being _officially maintained_, please **don't** archive this repository or do anything to it. I love this repo, and I still want to make discussions on it.

Also, having pull requests is cool, because hacks could be maintained by outside contributors.

## Chapter 6: Thank you

Shutting down or not, I'd like to say a humongous thank you to PMGH.

PMGH has successfully fucked Prodigy in the ass (just a tiny bit, no major damage, we're trying to raise awareness anyway) by giving skids some stuff and I'm glad they did.

I've had a great time being Support/"Customer Service" in the discord server, and I've also made one hack myself, `Get All Mythical Epics`.

**Anyways, Have a truly amazing day, I love you all, and thank you for your service, PMGH Team.**

____

### \- [gemsvidø](https://solo.to/gemsvido)

**Comment by BoredFishRE** [comment]
You make some great points, however, 

A. We're not college students, we're high schoolers in freshman or sophomore and at the youngest, a middle schooler. Taking money for hacks for a game is not ethical to me. I pay for the server, but I don't feel comfortable taking money for something like this. It would also give grounds for prodigy to sue us.

B. The repo isn't going anywhere. We're going to give instructions on how to use P-NP and host it themselves. Y'all can continue development, just without us.

Please if you're going to do the raid, make an alt. You will definitely get banned 😉

**Reply by afkvido** [reply]
> We're not college students, we're high schoolers in freshman or sophomore and at the youngest, a middle schooler.

Sorry, I didn't know that.

<br/>

> Taking money for hacks for a game is not ethical to me. I pay for the server, but I don't feel comfortable taking money for something like this. 

bruh.


<br/>

>  It would also give grounds for prodigy to sue us.

Ok, **that** I do understand.


<br/>


> The repo isn't going anywhere. We're going to give instructions on how to use P-NP and host it themselves. Y'all can continue development, just without us.

Tysm :)




<br/>


> Please if you're going to do the raid, make an alt. You will definitely get banned 😉

Prodigy alt? I have many 😉

**Reply by afkvido** [reply]
Anyways, thanks for responding!

**Reply by afkvido** [reply]
Raid update: I've covered all the Spinner and Wheeler NPCs
<img width="1417" alt="Screen Shot 2022-04-02 at 11 32 20 PM" src="https://user-images.githubusercontent.com/69060894/161414988-5f4dcd0b-fe8a-4695-b50b-340f17c42688.png">

**Reply by afkvido** [reply]
ALL NPCS IN LAMPLIGHT ARE COVERED.

<img width="1255" alt="Screen Shot 2022-04-02 at 11 45 24 PM" src="https://user-images.githubusercontent.com/69060894/161415382-1b90c9e6-1229-485b-815f-be4cce571527.png">

GET FUCKED, PRODIGY!

**Reply by afkvido** [reply]
(animalia)

**Comment by ghost** [comment]
The hacks are all
# Error 502
anyways

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/99303106/160511975-e34a2a3c-9837-4799-87bf-b00f05f41fb2.png)

**Reply by afkvido** [reply]
Hacks are down, and I don't think that they have plans to fix the hacks, because they're shutting down anyway.

**Comment by ghost** [comment]
Can we get the source code file for https://hacks.prodigyhacking.com?

**Reply by afkvido** [reply]
@1603191909, it's right here: https://github.com/Prodigy-Hacking/P-NP

**Comment by ghost** [comment]
i thought there was a 1.5mb file

**Reply by afkvido** [reply]
Nope. hacks.prodigyhacking.com automatically updates it's versions to Prodigy's versions.

That's one of the reasons why P-NP exists, (the second reason is to update hacks), 

Thats also the reason why the "clear your cache to fix" bug exists

**Reply by ghost** [reply]
okay, so how do i end up with the large file?

**Reply by afkvido** [reply]
you probbably downloaded the contents of "hacks.prodigyhacking.com".

you've downloaded game.min.js,  but PNP auctomatically updates the versions to prodigy's versions.

**Reply by ghost** [reply]
ok, that makes sense
so PNP modifies it according to some "rules?"

**Reply by afkvido** [reply]
Instead of loading Prodigy's `game.min.js`, PHEx redirects the loading of JavaScript code to `hacks.prodigyhacking.com`.
